pkg/util/pool/bucketed_pool.go

```diff
@@ -86,11 +87,15 @@ func (p *BucketedPool[T, E]) Put(s T) {
 		return // Ignore slices larger than the largest bucket
```

Comment:
Should we note here that we expect this will never happen given the `maxSize` check a few lines above and the fact that `maxSize` should be a power of two?